CMakeLists.txt

```diff
@@ -1826,15 +1824,18 @@ if (UNIX AND X64)
   find_library(liblz4 lz4)
   if (liblz4)
     message(STATUS "Found liblz4: ${liblz4}")
+    if (APPLE)
+      find_path(LZ4_INCLUDE_DIR lz4.h HINTS /usr/local/include /opt/homebrew/include)
+      find_path(LZ4_LIB_DIR liblz4.a HINTS /usr/local/lib /opt/homebrew/lib)
+      include_directories(${LZ4_INCLUDE_DIR})
```

Comment:
If find_path fails, looks like the result will be lz4.h-NOTFOUND (https://cmake.org/cmake/help/latest/command/find_path.html),  and it'll just fail in include_directories?

Reply:
Looks like it (though the result will be LZ4_INCLUDE_DIR-NOTFOUND).  I mirrored the libsnappy lines above.  The include_ and link_ can be wrapped in an if(), here and for libsnappy, with a warning if they fail.  I went ahead and did that with a shared function for both of these.  It is weird that we need the lib path one after find_library().